### ISSUE TITLE
device names more restrictive

### ISSUE THREAD

**Issue by jlekhter** [bwssytems/ha-bridge#276]
I'm not sure if anything changed on Amazon's side, but as of today it seems like all my device names that were previously called family room, foyer, office, living room stopped being found by Alexa. It would say "Sorry cant find a device or group called [device name] lights in Jerry's account". I had to rename all of the devices to family room lights, foyer lights, office lights, etc. for Alexa to find them.

Also, I have devices called amanda's lights and sophie's lights which worked for on, but not off commands or vice versa. After looking up what Alexa heard, I noticed that one of the commands had an apostrophe s, but the other didnt in Alexa's history.  I had to add another set of devices so that I would have a set with apostrophes and a set without in order to get things to work.

Anyone else experiencing similar issues? Has something changed with the way Alexa understands device names? Sorry if this isnt an HA-Bridge question, but I dont have any native Hue Lights so I thought I would ask here.

Thanks,

Jerry

**Comment by bwssytems** [comment]
Device naming will always be trial and error. Amazon keeps updating their recognition software and they seem to start reserving words. The bridge has no control over this.

**Comment by jlekhter** [comment]
Ok. Thanks for the response. It seems odd that a device name works for ON but doesn't for OFF. I'll work around it in the mean time.

**Comment by audiofreak9** [comment]
@jlekhter, have you checked the Alexa History to see what was interpreted based on what you said?
 http://alexa.amazon.com/spa/index.html#settings/dialogs

I found that voice training helped my Echos and family get along better too.  http://alexa.amazon.com/spa/index.html#settings/voicetraining

**Comment by jlekhter** [comment]
I have looked at the history. 

I had a light called "amelias lights" then one day it stopped recognizing the light.  In all honesty turning it off may have worked, but I was trying to turn it on and couldnt get it to work so I never tried the off before I started making changes.
In Alexa's history, the pattern came back "Turn on amelia's lights". I changed my light name to "amelia's lights" (with an apostrophe) and the light started to turn on. Then I noticed that I couldn't turn the light off. 

When I looked at Alexa's history, the pattern read "Turn off amelias lights". This time no apostrophe. 

So now I created a second device to control the same light. One had an apostrophe and the other didnt. Both active at the same time and Alexa seems to be finding them just fine.

The same day a bunch of other lights stopped responding too. 
"amelias LED lights" had to be changed to "amelias l. e. d. lights" -- notice the space between the periods.
"office" had to be changed to "office lights"
"guest room" had to be changed to "guest room lights"

In total, I probably had to change 15 or so lights and add 4 devices with apostrophe and without to make things work again.

I havent tried voice training -- but will see if that makes things better when I get home this weekend.
